build/azuredevops/azure-pipelines-release.yml

```diff
@@ -0,0 +1,143 @@
+                submodules: true
+                fetchTags: true
+                persistCredentials: True
+              - task: MicroBuildSigningPlugin@4
+                inputs:
+                  signType: '$(SignType)'
+                  feedSource: 'https://devdiv.pkgs.visualstudio.com/DefaultCollection/_packaging/MicroBuildToolset/nuget/v3/index.json'
+                env:
+                  TeamName: '$(TeamName)'
+              - task: NodeTool@0
+                displayName: Use Node 18.x
+                inputs:
+                  versionSpec: 18.x
+              - task: CmdLine@2
+                displayName: npm install
+                inputs:
+                  script: npm run install:all
+              - task: CmdLine@2
+                displayName: Package VSIX
+                inputs:
+                  script: |
+                    npm run package
+                    mv pyright-*.vsix $(VSIX_NAME)
+                  workingDirectory: packages/vscode-pyright
+              - task: CopyFiles@2
+                displayName: 'Copy vsix to: build_output'
+                inputs:
+                  SourceFolder: packages/vscode-pyright
+                  Contents: '*.vsix'
+                  TargetFolder: build_output
+      - stage: CreateRelease
+        dependsOn:
+          - BuildVsix
+        jobs:
+          - job: create_release
+            displayName: Create GitHub Release
+            steps:
+              - checkout: none
+              - download: current
+              - task: GitHubRelease@1 #https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/github-release-v1?view=azure-pipelines
+                displayName: 'Create GitHub Release'
+                inputs:
+                  gitHubConnection: 'Github-Pylance' # The name of the GitHub service connection
+                  repositoryName: 'microsoft/pyright' # The name of your GitHub repository
+                  action: 'create'
+                  isDraft: true
+                  isPreRelease: false
+                  addChangeLog: true
+                  title: 'Published $(Build.SourceBranchName)'
+                  assets: |
+                    $(Pipeline.Workspace)/$(ARTIFACT_NAME_VSIX)/$(VSIX_NAME)
+      - stage: WaitForValidation
+        dependsOn:
+          - CreateRelease
+        jobs:
+          - job: wait_for_validation
+            displayName: Wait for manual validation
+            pool: server
+            steps:
+              - task: ManualValidation@0
+                timeoutInMinutes: 1440 # task times out in 1 day
+                inputs:
+                  notifyUsers: 'plseng@microsoft.com'
+                  instructions: 'Please test the latest draft release of Pyright.'
+                  onTimeout: 'reject'
+
+      - stage: PublishExtension
+        dependsOn:
+          - WaitForValidation
+        jobs:
+          - job: publish_extension
+            displayName: Publish extension to marketplace

```

Comment:
Are you planning to eventually automate publishing to npm also?

Reply:
when Eric mentioned that today. i was going to take a look.